### ISSUE TITLE
[The term 'Start-AutoRestCodeGeneration' is not recognized as the name of a cmdlet to generate the SDK]

### ISSUE THREAD

**Issue by hari-bodicherla** [Azure/azure-sdk-for-net#7642]
**Query/Question**
Installed Autorest through NPM and ran the generate.ps1 under sdk\compute\Microsoft.Azure.Management.Compute\src. The script resulting in below error -
Start-AutoRestCodeGeneration : The term 'Start-AutoRestCodeGeneration' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try 
again.

***Why is this not a Bug or a feature Request?***
Because this is working for few others who forked the repo and ran the generate.ps1 script.

**Setup (please complete the following information if applicable):**
 - OS: Windows
 - IDE : [e.g. IntelliJ]
 - Version of the Library used
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Query Added
- [ ] Setup information Added

**Comment by fstugren** [comment]
I just ran into the same thing. I enlisted in the code, installed autorest, restored nuget packages etc. I can see the cmdled referenced in dozens of generate.ps1 scripts in the source code, but I couldn't find out how to install the powershell module with those cmdlets. Is there information in the GitHub documents associated with the Azure sdk that I missed?

**Comment by hari-bodicherla** [comment]
I guess the documentation is somehow scattered and not up to date. Can you try this steps in the below link. it worked for me to unblock that step atleast.
https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec

**Comment by fstugren** [comment]
Thanks, that worked, albeit in a convoluted way.
Interesting, running this command:
`msbuild myproject.csproj  /t:Util /p:UtilityName=InstallPsModules`
installed the missing PowerShell modules; then generate.ps1 worked just fine.
I would expect details like this to be on the ["contributing" page for the Azure SDK](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md):
and in fact a similar command is mentioned there:
`dotnet msbuild build.proj /t:Util /p:UtilityName=InstallPsModules`
however it's under the section "Non-Windows command line build" so I ignored it.

**Comment by hari-bodicherla** [comment]
Nice.. Will try to add this link or info for any TSG.  Also now I am facing the issue in generating the SDKs from the branch in my forked repo. Will create or update the issue to reflect that. Did you try to generate the SDK.

**Comment by fstugren** [comment]
Yes, I was able to generate the SDK. Eventually I figured out you can bypass the powershell commands and invoke the autorest command directly in order to have more control on the output during development.
